### PR TITLE
Automatically build docker terraform image

### PR DESCRIPTION
**Pull request by zzxwill**
Merge repo oam-dev/docker-terraform-base into this repo and automatically
build the image

Signed-off-by: Zheng Xi Zhou <zzxwill@gmail.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
     unzip terraform_${TERRAFORM_VERSION}_linux_amd64.zip -d /usr/bin
+COPY retrieve_tf_provider.sh /tmp
 
-ARG VCS_REF
+ENV RETRIEVE_TF_PROVIDER=/tmp/retrieve_tf_provider.sh
 
-LABEL org.label-schema.vcs-ref=$VCS_REF \
-    org.label-schema.vcs-url="https://github.com/broadinstitute/docker-terraform"
+RUN $RETRIEVE_TF_PROVIDER random 3.1.0
+RUN $RETRIEVE_TF_PROVIDER alicloud 1.140.0
```
Comment (lowkeyrd) [inline Dockerfile:25]: Maybe always use the newest version is better.

Comment (zzxwill) [reply Dockerfile:25]: alicloud or random?

File: Dockerfile [hunk 2]
```diff
@@ -1,17 +1,29 @@
-FROM oamdev/docker-terraform-base:1.0.9
+FROM alpine:3.13
+
+RUN \
+  apk update && \
+  apk add bash py-pip && \
+  apk add --virtual=build gcc libffi-dev musl-dev openssl-dev python3-dev make && \
+  apk add curl jq python3 ca-certificates git openssl unzip wget && \
+  pip --no-cache-dir install -U pip && \
+  pip install azure-cli && \
```
Comment (lowkeyrd) [inline Dockerfile:9]: Why used the azure-cli here?

**Review by lowkeyrd — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (zzxwill) [reply Dockerfile:9]: When the provide is `azurerm`, it's needed.

**Pull request merged**